astropy/modeling/core.py

```diff
@@ -2438,45 +2446,62 @@ def render_model(model, arr=None, coords=None):
     """
 
     bbox = model.bounding_box
+    ndim = model.n_inputs
 
     if (coords is None) & (arr is None) & (bbox is None):
-        raise AssertionError('If no bounding_box is set, coords or arr must be input.')
+        raise AssertionError('If no bounding_box is set, '
+                             'coords or arr must be input.')
 
     # for consistent indexing
-    if model.n_inputs == 1:
+    if ndim == 1:
         if coords is not None:
             coords = [coords]
```

Comment:
That particular case (single pixel) won't work with a bounding box, but in general yes because it expects a list-like of length `model.n_inputs` to iterate over (for n-dimensional generalization).